frontend/views/containers/user-settings/AppLogs.vue

```diff
@@ -0,0 +1,177 @@
+            i18n Errors
+          label.checkbox
+            input.input(type='checkbox' name='filter' v-model='form.filter' value='warn')
+            i18n Warnings
+          label.checkbox
+            input.input(type='checkbox' name='filter' v-model='form.filter' value='debug')
+            i18n Debug
+          label.checkbox
+            input.input(type='checkbox' name='filter' v-model='form.filter' value='info')
+            i18n Info
+          label.checkbox
+            input.input(type='checkbox' name='filter' v-model='form.filter' value='log')
+            i18n Log
+
+        button.is-small.c-download(@click='downloadLogs')
+          i.icon-download.is-prefix.c-icon
+          i18n.hide-touch Download
+          i18n.hide-desktop Share
+        a(ref='linkDownload' hidden)
+
+      textarea.textarea.c-logs(ref='textarea' rows='12' readonly)
+        | {{ prettyLogs }}
+
+      i18n.link(tag='a' @click='openTroubleshooting') Troubleshooting
+</template>
+
+<script>
+import { mapState, mapMutations, mapActions } from 'vuex'
+import sbp from '~/shared/sbp.js'
+import { CAPTURED_LOGS, SET_APP_LOGS_FILTER } from '@utils/events.js'
+import { downloadLogs } from '@model/captureLogs.js'
+export default {
+  name: 'AppLogs',
+  components: {},
+  data () {
+    return {
+      form: {
+        filter: []
+      },
+      ephemeral: {
+        logs: []
+      }
+    }
+  },
+  created () {
+    this.form.filter = this.appLogsFilter
+    sbp('okTurtles.events/on', CAPTURED_LOGS, this.addLog)
+
+    const logs = []
+    let lastEntry = localStorage.getItem('giConsole/lastEntry')
+    do {
+      const entry = JSON.parse(localStorage.getItem(`giConsole/${lastEntry}`))
+      if (!entry) break
+      logs.push(entry)
+      lastEntry = entry.prev
+    } while (lastEntry)
+    this.ephemeral.logs = logs.reverse()
+  },
+  beforeDestroy () {
+    sbp('okTurtles.events/off', CAPTURED_LOGS)
+  },
+  watch: {
+    'form.filter' (filters) {
+      this.setAppLogsFilters(filters)
+      sbp('okTurtles.events/emit', SET_APP_LOGS_FILTER, filters)
+    },
+    prettyLogs () {
+      // Automatically scroll textarea to the bottom
+      this.$nextTick(() => {
+        if (this.$refs.textarea) {
+          this.$refs.textarea.scrollTop = this.$refs.textarea.scrollHeight
+        }
+      })
+    }
+  },
+  computed: {
+    ...mapState([
+      'appLogsFilter'
+    ]),
+    prettyLogs () {
+      const isEntryNew = (type) => ['NEW_SESSION', 'NEW_VISIT'].includes(type)
+      return this.ephemeral.logs
+        .filter(({ type }) => {
+          return this.form.filter.includes(type) || isEntryNew(type)
+        })
+        .map(({ type, msg, timestamp }) => {
+          return isEntryNew(type)
+            ? `--------------- \n${timestamp} [${type}]`
+            : `${timestamp} [${type}] ${msg.map(part => JSON.stringify(part)).join(' ')}`
+        })
+        .join('\n')
+    }
```

Comment:
Per the other comment, this can be now simplified to:

```js
    prettyLogs () {
      return this.ephemeral.logs
        .filter(({ type }) => this.form.filter.includes(type))
        .map(({ type, msg, timestamp }) =>
          `${timestamp} [${type}] ${msg.map(JSON.stringify).join(' ')}`
        )
        .join('\n')
    }
```